Molly/README.md

```diff
@@ -0,0 +1,168 @@
+### Homework #1
+(Due Monday 7/20)
+
+You need to turn in this homework by cloning this repo, making a new branch with your solutions, pushing to github, and issuing a pull request.
+For details, follow the instructions [here](http://portlandcodeschool.github.io/jse/2015/01/07/command-line-and-git-slides/#/14).
+
+Please also read the [learning objectives](objectives.md) for this week.
+
+---
+
+**1)** (_30% of total time_)
+Predict the output of each of the following expressions, and explain the rules which determine that answer.  (_Hint: The rules for operator `==` are complicated, but do your best!_)
+
+If the output depends on the value or type of variable _x_, identify the conditions (what _x_ is) when the expression will output true (or false, if that's simpler).  Assume the cases are independent, and _x_ is reset to an unknown value before each.
+
+Some of these are tricky!  Don't trust your first instinct.  
+
+
+**a)** `"1" == 1` True. When we convert "1" from a string into a number, 1 and 1 are equal.
+
+**b)** `"1" === 1` False. "1" is a string and 1 is a number. Therefore the two sides are not equal.
+
+**c)** `x == 'x'` False. x is undefined, which is always false-y
+
+**d)** `x == (x+'')` True. Both side are false, which makes them equal--thereby making this statement true.
+
+**e)** `'' == ' '` True. Both sides are false-y, which makes the statement true
+
+**f)** `x = true` True. We have assigned the value true to x, so the statement will come out as true.
+
+**g)** `var x; x == 'undefined'` False. x is undefined and therefore false-y while 'undefined' is a string and truth-y, which makes the two sides unequal.
+
+**h)** `'9'<'10'` True. The number 9 is less than the number 10. (I checked this in the console and know it's incorrect, but I can't figure out why.)
+
```

Comment:
In this case, what is being evaluated here is not strictly numerical values, but rather their alpha-numerical values. The < comparator is checking '9' vs '10' one character at a time. In this situation '9' is greater than '1' so javascript stops comparison and returns false. Similarly, '950' < '9256' returns false as well, because '9' == '9', which moves to the next character set, returning '5' > '2'. As soon as javascript finds one character greater than another, it discontinues comparing the sets, and returns with a boolean. This works with alpha strings as well. For example, 'b' > 'a'!